Fixed issue 37

chrisbrownie opened the pull request:
Added -ReportFile parameter that permits the caller to provide a relative or absolute filename. If no filename is provided, the default will be utilised with a date/timestamp added.

cunninghamp commented:
Thanks for your contribution @chrisbrownie. Here's a few things I've noted in testing:

- Script help hasn't been updated for new parameter (examples should also include the fact that UNC path works)
- Filename is validated as .html. I can use this command and get into a bit of trouble when the file tries to open:

```
.\Run-ExchangeAnalyzer.ps1 -FileName foo
```

- File generation fails if output path doesn't exist

```
[PS] C:\Scripts\Exchange Analyzer>.\Run-ExchangeAnalyzer.ps1 -FileName C:\doesntexist\foo.html
Out-File : Could not find a part of the path 'C:\doesntexist\foo.html'.
At C:\Scripts\Exchange Analyzer\Run-ExchangeAnalyzer.ps1:413 char:15
+ $reportHtml | Out-File $reportFile -Force
+               ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OpenError: (:) [Out-File], DirectoryNotFoundException
    + FullyQualifiedErrorId : FileOpenFailure,Microsoft.PowerShell.Commands.OutFileCommand

C:\doesntexist\foo.html : The term 'C:\doesntexist\foo.html' is not recognized as the name of a cmdlet, function,
script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is
correct and try again.
At line:1 char:1
+ C:\doesntexist\foo.html
+ ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\doesntexist\foo.html:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
```

chrisbrownie commented:
Thanks for the feedback, @cunninghamp. I've amended the script in the following ways:
* Added PARAMETER and EXAMPLE sections to help for this new parameter
* Added array `$supportedOutputFileTypes` to Variables region to contain list of supported output times (allowing flexibility for xml or similar to be added in future)
* Added validation of file type to ensure extension is one of the supported types
* Added logic to create path if it does not exist, if possible, and to throw an exception otherwise
